core/client/fs/src/main/java/alluxio/client/file/RetryHandlingFileSystemMasterClient.java

```diff
@@ -91,23 +91,39 @@ protected void afterConnect() {
   @Override
   public synchronized List<AlluxioURI> checkConsistency(final AlluxioURI path,
       final CheckConsistencyOptions options) throws AlluxioStatusException {
-    return retryRPC(() -> {
-      List<String> inconsistentPaths =
-          mClient.checkConsistency(path.getPath(), options.toThrift()).getInconsistentPaths();
-      List<AlluxioURI> inconsistentUris = new ArrayList<>(inconsistentPaths.size());
-      for (String inconsistentPath : inconsistentPaths) {
-        inconsistentUris.add(new AlluxioURI(inconsistentPath));
-      }
-      return inconsistentUris;
+    return retryRPC(new RpcCallable<List<AlluxioURI>>() {
+      @Override
+      public List<AlluxioURI> call() throws TException {
+        List<String> inconsistentPaths =
+            mClient.checkConsistency(path.getPath(), options.toThrift()).getInconsistentPaths();
+        List<AlluxioURI> inconsistentUris = new ArrayList<>(inconsistentPaths.size());
+        for (String inconsistentPath : inconsistentPaths) {
+          inconsistentUris.add(new AlluxioURI(inconsistentPath));
+        }
+        return inconsistentUris;
+      }
+
+      @Override
+      public String getName() {
+        return "CheckConsistency";
```

Comment:
Do you think we can use reflection to get the method name?

Reply:
Hmm I wasn't able to find a good way, are you suggesting something like `Thread.currentThread().getStackTrace()`? Afaik there isn't a clean and efficient way of doing it, and the code here is in the critical path.

Reply:
With the additional perf overhead, and the frequency of method call, I think it's okay to go with your approach with some information duplication.